ui/src/importer/Importer.js

```diff
@@ -121,9 +132,10 @@ export class Importer extends Component {
           </Button>
           <Button
             primary
-            onClick={() => {
+            action="DELETE"
+            onClick={(e, props) => {
               this.setState({ openModal: false });
-              this.handleSubmit();
+              this.handleSubmit(e, props);
```

Comment:
Why not passing only the `action` here? No need for `e` and all props if I am not mistaken.

Reply:
I did it because the code is cleaner in any other cases of assigning this method to onClick event:
```
              <Button
                secondary
                content="Import"
                onClick={this.handleSubmit}
                action="IMPORT"
              />
```
where signature `handleSubmit = (e, { action })` is quite neat. In this particular case however there is one more step needed to set the modal state before calling handleSubmit. So my options were to pass all the props of the button or to map it before method call. The former seemed cleaner to me in terms of props manipulation. Would you prefer I change it?

Reply:
Ok I understand. The risk in general is that in the future someone will abuse of that, having all the props already there, and create some dirty code.
I would suggest to always first create the inner methods/funcs that will have, as params, only the minimum needed, nothing more (and ideally pure functions).

Reply:
👍  I will reimplement it